### PR TITLE
[stdlib][Part 1] De-gyb almost everything

### PR DESCRIPTION
**Pull request by Azoy**
See here: https://forums.swift.org/t/struggling-to-find-things-in-the-new-stdlib-for-5-0-e-g-where-is-addingreportingoverflow-implemented/18612/7

This is part 1 in a 4 part series.

This removes most `.gyb` files from the stdlib. Commits titled `de-gyb ...` are the product of copying and pasting the de-gybbed version over. I have a post de-gyb formatting commit that cleans up some of the messy formatting from gyb.

cc: @airspeedswift @stephentyrone @lorentey @xwu

### CODE REVIEW

**Comment by stephentyrone** [comment]
@airspeedswift and I just talked about this briefly. We'd like to do this, but want to be a bit more gradual about it. In particular, Ben wants to keep the `.gyb` files around (because they're better for editing--at least once you're used to them), but add a test that validates that the result of expanding the gyb matches the non-gyb source.

In cases where there isn't *too* much duplication, removing gyb entirely makes sense, but we should do those one file-per-commit, I think.

**Comment by Azoy** [comment]
Understandable. I read "all in" from Ben and I was ready to go "all in" 😅 I imagine we might one day get around to de-gybbing `IntegerTypes` which might allow us to break up the integers into separate files, `Int8.swift`, `Int16.swift`, etc. How would we go about testing this split in files with `IntegerTypes.swift.gyb`? (Assuming we want to do something like that instead of just testing against a `IntegerTypes.swift` with everything in it)

**Comment by stephentyrone** [comment]
@Azoy let's leave IntegerTypes and FloatingPointTypes alone for the time being, since they're a bit more involved than the others (but: we would simply run the test multiple times with different arguments to validate).

I think the other thing worth doing here is either to more clearly document *how* to generate the .swift files from the .swift.gyb (probably via a comment line at the top the .gyb file that has exactly the required invocation(s)).

**Comment by airspeedswift** [comment]
> which might allow us to break up the integers into separate files, `Int8.swift`, `Int16.swift`

I'd rather leave them all in one big file than clutter the folder with lots of near-identical files. The goal of degybbing is to improve search ability of the source code (the "where on earth is xxx declared" question on the forum that led to this idea), to allow stepping through the std lib, allow an LSP server to parse the whole std lib etc, all of which works from the single file approach. These files are probably rarely going to be touched in future so making them easier to navigate/read for editing purposes isn't that important.

**Comment by Azoy** [comment]
I only bring up splitting them up because the gyb file itself is almost 2k LoC, and a single ungybbed `IntegerTypes.swift` would be about 20k LoC. Some editors/IDEs/github even would have some trouble handling a buffer that large which might decrease ability to efficiently search for source code.

**Comment by stephentyrone** [comment]
@Azoy @airspeedswift Also worth noting that as the generics story and compiler optimizations get better, we can move progressively more and more integer stuff out of gyb and into default implementations, so the gyb piece will gradually get smaller.

File: stdlib/public/core/Runtime.swift [hunk 1]
```diff
@@ -169,9 +221,70 @@ internal func _swift_stdlib_atomicFetchAddInt(
 /// A 32 byte buffer.
 internal struct _Buffer32 {
   internal init() {}
-% for i in range(32):
-  internal var _x${i}: UInt8 = 0
-% end
+
```
Comment (airspeedswift) [inline stdlib/public/core/Runtime.swift]: tiny nit... can we remove all the spacing between the vars here and below?

**Review by airspeedswift — APPROVED** [review]

File: stdlib/public/core/BuiltinMath.swift.gyb [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-        return 'f'
-    if bits == 64:
-        return ''
-    if bits == 80:
-        return 'l'
-
-# Each of the following lists is ordered to match math.h
-
-# These functions have a corresponding LLVM intrinsic
-# Note, keep this up to date with Darwin/tgmath.swift.gyb
```
Comment (xwu) [inline stdlib/public/core/BuiltinMath.swift.gyb]: I think these are valuable comments that should be preserved in some way. They disappear in the generated file. I’d imagine there are other such comments elsewhere.

Comment (Azoy) [reply stdlib/public/core/BuiltinMath.swift.gyb]: I've gone through every file that's changing here looking for any gyb comments, and while there wasn't much, I've added those comments back in.

**Comment by Azoy** [comment]
While looking for gyb comments in `ExistentialCollection.swift`, I made some more minor formatting changes which might need a quick review.

**Comment by CodaFi** [comment]
@Azoy Could you rebase this one more time?

**Comment by Azoy** [comment]
@CodaFi I've successfully rebased! @airspeedswift @stephentyrone I know it's been a WHILE, but I would love another look to make sure this is good to go!

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@CodaFi *sigh* sorry bout that one... it should be good now.

**Comment by CodaFi** [comment]
It happens. Let’s give it another go.

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
Ignore the macOS result. There’s an infrastructure problem here.

**Comment by Azoy** [comment]
@CodaFi I've rebased and now this should be ready.

**Comment by CodaFi** [comment]
@swift-ci please test

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by Azoy** [comment]
macOS failure unrelated?

**Comment by CodaFi** [comment]
Ugh, infrastructure failure.  Let's try again

@swift-ci please test macOS platform

**Comment by Azoy** [comment]
Another infrastructure problem again? x)

**Comment by CodaFi** [comment]
I think this one’s an actual failure

> api-digester/dump-module.swift

Suggests you might have changed the ABI surface of stdlib.

**Comment by benrimmington** [comment]
The conformances of `Int` (to `MirrorPath`, `CVarArg`, and `CustomReflectable`) in the `cake` test module have been *reordered*, possibly due to your CMakeLists.txt changes.

There's a comment in [dump-module.swift](https://github.com/apple/swift/blob/9dab211cfb58e04d91b18826d540e685b6d79521/test/api-digester/dump-module.swift#L17-L19) which might be relevant:

> // The input JSON files need to be modified when standard library declarations
> // are reordered. This is expected behavior and we simply shouldn't run the test
> // when automatically evolving the standard library.

File: stdlib/public/core/CMakeLists.txt [hunk 3]
```diff
@@ -61,6 +63,7 @@ set(SWIFTLIB_ESSENTIAL
   EmptyCollection.swift
   Equatable.swift
   ErrorType.swift
+  ExistentialCollection.swift
```
Comment (benrimmington) [inline stdlib/public/core/CMakeLists.txt]: This won't fix the test failure, but you could move ExistentialCollection.swift to the SWIFTLIB_SOURCES list (at line 210) if it isn't an "essential" source; and also change the copyright year of CMakeLists.txt (from 2018 to 2019).

**Comment by Azoy** [comment]
@benrimmington thanks so much for the help.
@CodaFi I've tested locally and this test passes now. We should be good to go.

**Comment by theblixguy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@CodaFi all tests have passed!

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**